src/LetsTrace.Jaeger/Serialization/JaegerThriftSerialization.cs

```diff
@@ -84,10 +84,50 @@ public static List<JaegerTag> ConvertLogToJaegerTags(LogRecord log)
 
             foreach (var field in log.Fields)
             {
-                field.Marshal(tags);
+                AddToJaegerTagList(field.Key, field.Value, tags);
             }
 
             return tags;
         }
+
+        public static void AddToJaegerTagList(string key, object value, List<JaegerTag> tags)
+        {
+            switch(value)
+            {
+                case byte[] val:
+                    tags.Add(new JaegerTag{ Key = key, VType = JaegerTagType.BINARY, VBinary = val });
+                    break;
+                case string val:
+                    tags.Add(new JaegerTag{ Key = key, VType = JaegerTagType.STRING, VStr = val });
+                    break;
+                case double val:
+                    tags.Add(new JaegerTag{ Key = key, VType = JaegerTagType.DOUBLE, VDouble = val });
+                    break;
+                case decimal val:
+                    tags.Add(new JaegerTag{ Key = key, VType = JaegerTagType.DOUBLE, VDouble = (double)val });
+                    break;
+                case bool val:
+                    tags.Add(new JaegerTag{ Key = key, VType = JaegerTagType.BOOL, VBool = val });
+                    break;
+                case ushort val:
+                    tags.Add(new JaegerTag { Key = key, VType = JaegerTagType.LONG, VLong = val });
+                    break;
+                case uint val:
+                    tags.Add(new JaegerTag { Key = key, VType = JaegerTagType.LONG, VLong = val });
+                    break;
+                case ulong val:
+                    tags.Add(new JaegerTag{ Key = key, VType = JaegerTagType.LONG, VLong = (long)val });
```

Comment:
This cast will be a problem if the `ulong` is actually larger than the positive part of a `long`. I don't know if that throws an exception or just results in a different value.

Not sure if it's better to just not support `ulong` values?!

Reply:
It throws `Message: System.OverflowException : Value was either too large or too small for an Int64`

Reply:
I'd rather not get rid of it...what if we converted it to a string?

Reply:
yep, sounds good.